Examples/restcomm-olympus-nopod/restcomm-olympus/Utils.m

```diff
@@ -44,139 +46,252 @@ + (void) setupUserDefaults
+        if (contactsArray != nil){
+            //return non deleted contacts
+            NSArray *filterdArray = [Utils getFilteredContactArray:contactsArray];
+            return (int)[filterdArray count];
+        }
     }
+    return 0;
+}
+
+
++ (void)addContact:(LocalContact *)contact
+{
+    NSUserDefaults *appDefaults = [NSUserDefaults standardUserDefaults];
     
-    NSMutableDictionary * messages = [[appDefaults dictionaryForKey:@"chat-history"] mutableCopy];
-    if ([messages objectForKey:sipUri]) {
-        aliasMessages = [[messages objectForKey:sipUri] mutableCopy];
+    NSMutableArray * mutable = nil;
+    NSData *contactsArrayData = [appDefaults objectForKey:@"contacts"];
+    if (contactsArrayData != nil) {
+        NSArray *contactsArray = [NSKeyedUnarchiver unarchiveObjectWithData: contactsArrayData];
+        if (contactsArray != nil){
+            mutable = [contactsArray mutableCopy];
+        } else {
+            // should never happen
+            return;
+        }
+    } else {
+        // should never happen
+        return;
+    }
+    BOOL exists = NO;
+    //check if object is already added
+    for (int i=0; i < mutable.count; i ++){
+        LocalContact *savedContact = mutable[i];
+        if ([savedContact isEqual:contact]){
+            exists = YES;
+            break;
+        }
     }
     
-    [aliasMessages addObject:[NSDictionary dictionaryWithObjectsAndKeys:text, @"text", type, @"type", nil]];
-    [messages setObject:aliasMessages forKey:sipUri];
+    if (!exists){
+        [mutable addObject:contact];
+    }
     
-    [appDefaults setObject:messages forKey:@"chat-history"];
+    // update user defaults
+    [appDefaults setObject:[NSKeyedArchiver archivedDataWithRootObject:mutable] forKey:@"contacts"];
 }
 
-+ (NSArray*)contactForIndex:(int)index
++ (void)removeContactAtIndex:(int)index
 {
+    //contact will have the flag deleted set to true
     NSUserDefaults* appDefaults = [NSUserDefaults standardUserDefaults];
-    NSArray * contacts = [appDefaults arrayForKey:@"contacts"];
-    if (contacts) {
-        if ([contacts count] > index) {
-            return [contacts objectAtIndex:index];
+    NSMutableArray * mutable = nil;
+    NSData *contactsArrayData = [appDefaults objectForKey:@"contacts"];
+    
+    if (contactsArrayData != nil) {
+        NSArray *contactsArray = [NSKeyedUnarchiver unarchiveObjectWithData: contactsArrayData];
+        if (contactsArray != nil){
+            NSArray *filterdArray = [Utils getFilteredContactArray:contactsArray];
+            LocalContact *localContact = [filterdArray objectAtIndex:index];
+            
+            mutable = [contactsArray mutableCopy];
+            for (int i=0; i<mutable.count; i ++){
+                LocalContact *fromNonFilteredContact = [mutable objectAtIndex:i];
+                if ([fromNonFilteredContact isEqual:localContact]){
+                    fromNonFilteredContact.deleted = YES;
```

Comment:
When we delete an entry which is not local (its from device), we are deleting it only for Olympus app. So, the next time user starts the app, we will add it again (but user want it deleted).
When we have the delete flag, we know that user doesn't want it when we import phones again.

Reply:
I see, makes sense. But are you sure that the contacts are imported in Olympus whenever the App starts? Because I recall opening Olympus, then leaving it, going to phone contacts and adding a brand new contact, then opening Olympus again, but the contact was missing